tests/openfl/component/assigner/test_custom_assigner.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""CustomAssigner tests."""
+
+from unittest import mock
+
+import pytest
+
+from openfl.component.aggregation_functions import GeometricMedian
+from openfl.component.aggregation_functions import WeightedAverage
+from openfl.component.assigner.custom_assigner import Assigner
+from openfl.component.assigner.tasks import TrainTask
+from openfl.component.assigner.tasks import ValidateTask
+
+
+def assigner_function(collaborators, round_number, **kwargs):
+    """Return tasks by collaborator."""
+    tasks_by_collaborator = {}
+    for collaborator in collaborators:
+        tasks_by_collaborator[collaborator] = [
+            TrainTask(
+                name='train',
+                function_name='train_func',
+            ),
+            ValidateTask(
+                name='locally_tuned_model_validate',
+                function_name='validate',
+                apply_local=True,
+            ),
+            ValidateTask(
+                name='aggregated_model_validate',
+                function_name='validate',
+            ),
+        ]
+    return tasks_by_collaborator
+
+
+@pytest.fixture()
+def assigner():
+    """Return Assigner fixture."""
+    assigner = Assigner(
+        assigner_function=assigner_function,
+        aggregation_functions_by_task={
+            'train_func': GeometricMedian()
+        },
+        authorized_cols=['one', 'two'],
+        rounds_to_train=10,
+    )
+    assigner.define_task_assignments = mock.Mock()
+    return assigner
+
+
+def test_define_task_assignments(assigner):
+    """Test `define_task_assignments` is working."""
+    assigner.define_task_assignments()
+
+
+def test_get_tasks_for_collaborator(assigner):
+    """Test `get_tasks_for_collaborator` base working."""
+    tasks = assigner.get_tasks_for_collaborator('one', 2)
+
+    assert len(tasks) == 3
+    assert isinstance(tasks[0], TrainTask)
+    assert isinstance(tasks[1], ValidateTask)
+
+
+def test_get_collaborators_for_task(assigner):
+    """Test `get_collaborators_for_task` base working."""
+    collaborators = assigner.get_collaborators_for_task('train', 2)
+
+    assert collaborators == ['one', 'two']
+
+
+def test_get_all_tasks_for_round(assigner):
+    """Test `get_all_tasks_for_round` base working."""
+    all_tasks = assigner.get_all_tasks_for_round(2)
+
+    assert len(all_tasks) == 3
```

Comment:
Should we also not check the task names here as well?

Reply:
Added